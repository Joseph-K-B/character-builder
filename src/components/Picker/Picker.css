.image-controls{
    background-color: crimson;
}