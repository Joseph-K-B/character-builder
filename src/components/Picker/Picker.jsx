import React from "react";

function Picker({
    head,
    setHead
}) 
// const [head, setHead] = useState()
{
    const heads = ['dog', 'cat']
    return (
        <section>
            <div>
                <select name='head' key='head-select' value={head} onChange={(e) => setHead(e.target.value)}>
                    {heads.map((pic) => (
                        <option key={pic}>{pic}</option>
                    ))}
                </select>
            </div>
        </section>
    )
}

export default Picker