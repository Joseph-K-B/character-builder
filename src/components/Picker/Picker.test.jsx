import { render, screen } from "@testing-library/react"
import Picker from "./Picker"

it('should render picker component', () => {
    const pres = render(<Picker />);
    const selectHead = screen.getByLabelText(/Head/);
    expect(selectHead).toBeInTheDocument();
    const selectTorso = screen.getByLabelText(/Torso/);
    expect(selectTorso).toBeInTheDocument();
    const selectBottom = screen.getByLabelText(/Bottom/);
    expect(selectBottom).toBeInTheDocument();
    expect(pres).toMatchSnapshot();
})