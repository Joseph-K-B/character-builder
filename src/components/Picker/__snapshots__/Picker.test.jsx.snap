// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render picker component 1`] = `
<div>
  <section
    class="image-controls"
  >
    <div
      class="head-control"
    >
      <label
        class="head-label"
        for="head"
      >
        Head
      </label>
      <select
        class="select"
        id="head"
        name="head"
      >
        <option>
          dog
        </option>
        <option>
          dog-2
        </option>
        <option>
          dog-3
        </option>
        <option>
          dog-4
        </option>
        <option>
          dog-5
        </option>
        <option>
          horse
        </option>
        <option>
          cat
        </option>
        <option>
          cat-2
        </option>
        <option>
          cat-3
        </option>
      </select>
    </div>
    <div
      class="torso-control"
    >
      <label
        for="torso"
      >
        Torso
      </label>
      <select
        class="select"
        id="torso"
        name="torso"
      >
        <option>
          leather
        </option>
        <option>
          suit
        </option>
        <option>
          gray
        </option>
        <option>
          firefighter
        </option>
        <option>
          straight
        </option>
      </select>
    </div>
    <div
      class="bottom-control"
    >
      <label
        for="torso"
      >
        Bottom
      </label>
      <select
        class="select"
        id="bottom"
        name="bottom"
      >
        <option>
          jeans
        </option>
        <option>
          splatter
        </option>
        <option>
          sweat
        </option>
        <option>
          jort
        </option>
      </select>
    </div>
  </section>
</div>
`;
